pkg/validator/validators.go

```diff
@@ -182,6 +246,54 @@ func egressValidator(req *admissionv1.AdmissionRequest) (*admissionv1.AdmissionR
 	return nil, nil
 }
 
+// upstreamTrafficSettingValidator validates the UpstreamTrafficSetting custom resource
+func (kc *policyValidator) upstreamTrafficSettingValidator(req *admissionv1.AdmissionRequest) (*admissionv1.AdmissionResponse, error) {
+	upstreamTrafficSetting := &policyv1alpha1.UpstreamTrafficSetting{}
+	if err := json.NewDecoder(bytes.NewBuffer(req.Object.Raw)).Decode(upstreamTrafficSetting); err != nil {
+		return nil, err
+	}
+
+	hostComponents := strings.Split(upstreamTrafficSetting.Spec.Host, ".")
+	if len(hostComponents) < 2 {
+		return nil, field.Invalid(field.NewPath("spec").Child("host"), upstreamTrafficSetting.Spec.Host, "invalid FQDN specified as host")
+	}
+
+	err := kc.checkDuplicateHosts(upstreamTrafficSetting)
+	if errors.Is(err, errUpstreamTrafficSettingDuplicateHosts) {
+		return nil, err
+	}
+
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Let's replace this with a check `GetUpstreamTrafficSetting(opt)` where opt is of type UpstreamTrafficSettingGetOpt and supports lookups in the case based on host. This means you need to add a new field `Host` to the `UpstreamTrafficSettingGetOpt` type.

Once done, this can be simplified to:
```
if obj := policyController.GetUpstreamTrafficSetting(opt); obj != nil {
    // duplicate
}
```